add-dbt-mcp dev blog (#7218)
fishtown-analytics/docs.getdbt.com: website/blog/2025-04-18-dbt-mcp-server.md

---
title: "Introducing the dbt MCP Server – Bringing Structured Data to AI Workflows and Agents"
description: "We’re open‑sourcing an experimental dbt MCP server so LLMs and agents can discover, query, and run your dbt project."
slug: introducing-dbt-mcp-server
authors: [jason_ganz]
tags: [ai, data ecosystem]
hide_table_of_contents: false
date: 2025-04-21
is_featured: true
---

# Introducing the dbt MCP Server - Bringing Structured Data to AI Workflows and Agents

> dbt is the standard for creating governed, trustworthy datasets on top of your structured data. [MCP](https://www.anthropic.com/news/model-context-protocol) is [showing](https://techcrunch.com/2025/04/09/google-says-itll-embrace-anthropic-standard-for-connecting-ai-models-to-data/) [increasing](https://techcrunch.com/2025/03/26/openai-adopts-rival-anthropic-standard-for-connecting-ai-models-to-data/) [promise](https://www.microsoft.com/en-us/microsoft-copilot/blog/copilot-studio/introducing-model-context-protocol-mcp-in-copilot-studio-simplified-integration-with-ai-apps-and-agents/) as the standard for providing context to LLMs to allow them to function at a high level in real world, operational scenarios.

Reviewer: Links at showing and increasing not working. It's also not obvious in the ui that you're linking to 3 different articles.

Today, we are open sourcing an experimental version of the [dbt MCP server](https://github.com/dbt-labs/dbt-mcp/tree/main). We expect that over the coming years, structured data is going to become heavily integrated into AI workflows and that dbt will play a key role in building and provisioning this data.

> In particular, we expect both [Business Intelligence](https://roundup.getdbt.com/p/how-ai-will-disrupt-bi-as-we-know) and [Data Engineering](https://roundup.getdbt.com/p/how-ai-will-disrupt-data-engineering) will be driven by AI operating on top of the context defined in your dbt Projects.

Reviewer: ```suggestion
In particular, we expect both [Business Intelligence](https://roundup.getdbt.com/p/how-ai-will-disrupt-bi-as-we-know) and [Data Engineering](https://roundup.getdbt.com/p/how-ai-will-disrupt-data-engineering) will be driven by AI operating on top of the context defined in your dbt projects.
```

We believe it is important for the industry to start coalescing on best practices for safe and trustworthy ways to access your business data via LLM.

**What is MCP?**

MCP stands for Model Context Protocol - it is an open protocol released by Anthropic in [November of last year](https://www.anthropic.com/news/model-context-protocol) to allow AI systems to dynamically pull in context and data. Why does this matter?

> 
> 
> 
> Even the most sophisticated models are constrained by their isolation from data—trapped behind information silos and legacy systems. Every new data source requires its own custom implementation, making truly connected systems difficult to scale.
> 
> MCP addresses this challenge. It provides a universal, open standard for connecting AI systems with data sources, replacing fragmented integrations with a single protocol. - Anthropic
> 

> Since then, MCP has become widely supported, with Google, Microsoft and OpenAI all committing to support MCP. 

Reviewer: This is a little repetitive. I wonder if just tweaking this drives your point better? 
```suggestion
Since then, MCP has become widely supported, with Google, Microsoft and OpenAI all committing to their support of MCP. 
```

*Chat use case:*

We suggest using Claude Desktop for this and creating a custom [project](https://www.anthropic.com/news/projects) that includes a prompt explaining the use cases you are looking to cover. 

To get this working:

- Follow the instructions in the Readme to install the MCP server
- Validate that you have added the MCP config to your Claude desktop config. You should see ‘dbt’ when you go to Claude→Settings→Developer
<Lightbox src="/img/blog/2025-04-18-dbt-mcp-server/claudedesktop_settings_dbt_mcp.png" title="Claude Desktop – MCP server running in Developer settings" width="100%" />

- Create a new project called “analytics”. Give it a description of how an end user might interact with it.

<Lightbox src="/img/blog/2025-04-18-dbt-mcp-server/claudedesktop_project_card.png" title="Example Claude Desktop project connected to the dbt MCP server" width="100%" />

- **Add a custom prompt explaining that questions in this project will likely be routed through the dbt MCP server.** You’ll likely want to customize this to your particular organizational context.

>     - Ex: This conversation is connected to and knows about the information in your dbt Project via the dbt MCP server. When you receive a question that plausibly needs data from an external data source, you will likely want to use the tools available via the dbt MCP server to provide it.

Reviewer: ```suggestion
    - For example: This conversation is connected to and knows about the information in your dbt Project via the dbt MCP server. When you receive a question that plausibly needs data from an external data source, you will likely want to use the tools available via the dbt MCP server to provide it.
```
